term/src/terminalstate.rs

```diff
@@ -674,6 +674,19 @@ impl TerminalState {
         }
     }
 
+    pub fn erase_buffer(&mut self) {
+        self.erase_in_display(EraseInDisplay::EraseScrollback);
+
+        let row_index = self.screen.phys_row(self.cursor.y);
+        let row = self.screen.lines[row_index].clone();
+
+        self.erase_in_display(EraseInDisplay::EraseDisplay);
+
+        self.screen.lines[0] = row;
+
+        self.cursor.y = 0;
+    }
```

Comment:
Thanks for the review! This is a great idea, I didn't realize that actions could be configured for enum variants that have associated data. I have one question about the implementation of the enum for the mode. It seems that the enum needs to be (de)serializable, which is why I defined it in the `config::keyassignment` module, but the `term` crate isn't associated with the `config` crate since it is supposed to be standalone. Would you like for me to keep `erase_buffer` as a function or should I define another enum in this `terminalstate` module which the enum in `cofig::keyassignment` would map to?

Reply:
I'd be fine with keeping this new function, although I think we could make its name more closely match the enum variant (perhaps: `erase_scrollback_and_viewport`?).

The other alternatives are OK, but feel like a lot of effort relative to the reward.